packages/sanity/src/form/studio/FormProvider.tsx

```diff
@@ -74,7 +80,20 @@ export function FormProvider(props: FormProviderProps) {
     value,
   } = props
 
-  const {file, image, renderField, renderInput, renderItem, renderPreview} = useSource().form
+  const {file, image} = useSource().form
+
+  const Input = useInputComponent()
+  const Field = useFieldComponent()
+  const Preview = usePreviewComponent()
+  const Item = useItemComponent()
```

Comment:
These may be stored in context in the future, as an perf optimization. Maybe add a `NOTE` on that here?